### PR TITLE
Source verification 0.2.0-beta

### PR DESCRIPTION
**Pull request by edisinovcic**

### CODE REVIEW

File: plugins/source-verifier/profile.json [hunk 1]
```diff
@@ -2,13 +2,16 @@
...
     "description": "Source metadata fetcher and validator",
-    "version": "0.1.7-beta",
+    "version": "0.1.9-beta",
     "methods": [
      "fetch",
-     "verify"
+     "fetchByNetwork",
+     "verify",
+     "verifyByNetwork",
+     "verifyByForm"
```
Comment (yann300) [inline plugins/source-verifier/profile.json:11]: there is no `fetchAndSave(address: string, chain: any)` ?

File: plugins/source-verifier/api.ts [hunk 2]
```diff
@@ -1,16 +1,58 @@
...
+    
+    /**
+     * verify a contract. the network is retrieved from the application context.
+     * @param {verifyData} - map containing a metadata.json, source files, address and chain
+     * @return {VerifyResponse}
+     */
+    verify(address: string, files: any): VerifyResult
+    
+     /**
+     * Verify a contract. the network is retrieved from the application context.
```
Comment (yann300) [inline plugins/source-verifier/api.ts:51]: > the network is retrieved from the application context

in that case, this is by network id

File: plugins/source-verifier/profile.json [hunk 3]
```diff
@@ -2,13 +2,16 @@
     "name": "source-verification",
     "displayName": "Source Verifier",
     "description": "Source metadata fetcher and validator",
-    "version": "0.1.7-beta",
+    "version": "0.2.0-beta",
```
Comment (yann300) [inline plugins/source-verifier/profile.json:5]: In case you have some documentation, you can put a link here. The key is `documentation`

Comment (edisinovcic) [reply plugins/source-verifier/profile.json:5]: Currently closest to documentation for source verify is medium article which in my opinion is not the best resource to put here.

File: plugins/source-verifier/api.ts [hunk 4]
```diff
@@ -1,16 +1,58 @@
...
+export interface VerifyResult {
+  match: 'partial' | 'perfect' | 'no match',
+  success: boolean,
+  message: string
+}
+
+export interface VerifyData {
+  address: string,
+  chain: string | number,
+  files: Record<string, any>
```
Comment (yann300) [inline plugins/source-verifier/api.ts:15]: one of the files is the metadata.json right?

Comment (edisinovcic) [reply plugins/source-verifier/api.ts:15]: Yes

**Pull request merged**